modules/Kconfig.cmsis_nn

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2021, Commonwealth Scientific and Industrial Research
+# Organisation (CSIRO) ABN 41 687 119 230.
+# SPDX-License-Identifier: Apache-2.0
+
+comment "Components"
+
+config CMSIS_NN_ACTIVATION
+	bool "Activation"
+	default y
```

Comment:
Why is this and all settings below `default y` ?
Especially because it's opposite to the CMSIS-DSP lib where almost all components defaults to `n`:
https://github.com/zephyrproject-rtos/zephyr/blob/727ea490292255e3d0462f583fbc490a5c4936e4/modules/Kconfig.cmsis_dsp#L29
https://github.com/zephyrproject-rtos/zephyr/blob/727ea490292255e3d0462f583fbc490a5c4936e4/modules/Kconfig.cmsis_dsp#L43
https://github.com/zephyrproject-rtos/zephyr/blob/727ea490292255e3d0462f583fbc490a5c4936e4/modules/Kconfig.cmsis_dsp#L60

etc.

Reply:
They were `default y` in the old PR. I have removed this and added a new symbol `CMSIS_NN_ALL` which selects all the other symbols for convenience, similar to how the CMSIS-DSP options work.

Reply:
> They were `default y` in the old PR. I have removed this and added a new symbol `CMSIS_NN_ALL` which selects all the other symbols for convenience, similar to how the CMSIS-DSP options work.

maybe I missed it, but for CMSIS-DSP I only see that `_ALL` selection is available for the fast and fft tables.
Not library selection as is proposed in this PR.

Reply:
Right, but AFAIK CMSIS-NN is a backend that Tensorflow compiles down to, it seems unlikely that Tensorflow would only require an activation function for example. Manually specifying all of the options, if they are all going to be needed, will be annoying. Hence the helper symbol.

Reply:
so you say sensorflow is the main use-case for `CMSIS-NN` ?

So is TensorFlow expected to be selecting `select CMSIS_NN_ALL` ?
I'm simply asking to get a better understanding of the use-case, and why having a:
```
config SOMETHING
    select CMSIS_NN_ALL

+

config CMSIS_NN_ALL
    select n-lines
```

is better than
```
config SOMETHING
    select n-lines
```
directly.

Reason for being a little careful with such helper symbols is because they may easily end up in the stuck-symbol syndrome.

Reply:
The helper symbol has been removed, all configs must be individually selected